lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -0,0 +1,403 @@
+                If the port number is omitted, the default swarm listening port
+                is used.
+        default: 0.0.0.0:2377
+    force:
+        description:
+            - Use with state C(init) to force creating a new Swarm, even if already part of one.
+            - Use with state C(leave) to Leave the swarm even if this node is a manager.
+        type: bool
+        default: false
+    state:
+        description:
+            - Set to C(init), to create a new cluster.
+            - Set to C(join), to join an existing cluster.
+            - Set to C(leave), to leave an existing cluster.
+            - Set to C(remove), to remove a node.
+        required: true
+        default: join
+        choices:
+          - init
+          - join
+          - leave
+          - remove
+    node_id:
+        description:
+            - Used with I(state=remove). Swarm id of the node to remove.
+    join_token:
+        description:
+            - Used with I(state=join). Swarm token used to join a swarm cluster.
+    remote_addrs:
+        description:
+            - Used with I(state=join). Remote address of a manager to connect to.
+    task_history_retention_limit:
+        description:
+            - Maximum number of tasks history stored.
+    snapshot_interval:
+        description: Number of logs entries between snapshot.
+    keep_old_snapshots:
+        description: Number of snapshots to keep beyond the current snapshot.
+    log_entries_for_slow_followers:
+        description: Number of log entries to keep around to sync up slow followers after a snapshot is created.
+    heartbeat_tick:
+        description: Amount of ticks (in seconds) between each heartbeat.
+    election_tick:
+        description: Amount of ticks (in seconds) needed without a leader to trigger a new election.
+    dispatcher_heartbeat_period:
+        description: The delay for an agent to send a heartbeat to the dispatcher.
+    node_cert_expiry:
+        description: Automatic expiry for nodes certificates.
+    name:
+        description: Swarm's name
+    labels:
+        description: User-defined key/value metadata.
+    signing_ca_cert:
+        description: The desired signing CA certificate for all swarm node TLS leaf certificates, in PEM format.
+    signing_ca_key:
+        description: The desired signing CA key for all swarm node TLS leaf certificates, in PEM format.
+    ca_force_rotate:
+        description: An integer whose purpose is to force swarm
+                to generate a new signing CA certificate and key, if none have
+                been specified.
+    autolock_managers:
+        description: If set, generate a key and use it to lock data stored on the managers.
+        type: bool
+
+
+extends_documentation_fragment:
+    - docker
+
+requirements:
+    - "python >= 2.7"
+    - "Docker API >= 1.35"
+
+author:
+  - Thierry Bouvet (@tbouvet)
+'''
+
+EXAMPLES = '''
+
+- name: Init a new swarm
+  docker_swarm:
+    state: "init"
```

Comment:
No need to use quotes. We tend to not quote when it's not needed, as this helps people understand the YAML rules better.